### ISSUE TITLE
BasicExample fails to compile

### ISSUE THREAD

**Issue by samcragg** [samcragg/Crest#2]
The `GenerateOpenApi` target is causing the build to fail:

    Unhandled Exception: System.IO.FileLoadException: Could not load file or assembly 'System.Threading, Version=4.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)
       at System.Console.get_Out()
       at Microsoft.Extensions.CommandLineUtils.CommandLineApplication..ctor(Boolean throwOnUnexpectedArg)
       at Crest.OpenApi.Generator.Program.Main(String[] args)
    \Crest\example\BasicExample\BasicExample.csproj(22,5): error MSB3073: The command "dotnet crest-openapi \Crest\example\BasicExample\bin\Debug\netcoreapp2.0\BasicExample.dll" exited with code -532462766.

This seems to have been introduced when the OpenApi generator was upgraded to .NET Core 2.0 or it may be that CLI tools are only now supported by package reference instead of project reference

**Comment by samcragg** [comment]
Just experimenting with a blank solution whilst away from my home PC, ensure the tool project generates the NuGet package on build:

```
<PropertyGroup>
    <GeneratePackageOnBuild>true</GeneratePackageOnBuild>
</PropertyGroup>
```

Then add a [NuGet.config](https://docs.microsoft.com/en-gb/nuget/consume-packages/configuring-nuget-behavior) file to the consuming project to specify additional package sources as the build output:

```
NuGet.config
<configuration>
  <packageSources>
    <add key="Tool (Debug)" value="../tool/bin/Debug" />
    <add key="Tool (Release)" value="../tool/bin/Release" />
  </packageSources>
</configuration>
```

Finally make sure in the solution that the project dependencies are added so that the tool will build before the other project. You can then use the `DotNetCliToolReference` to reference the tool instead of doing a project reference (which doesn't work)

**Comment by samcragg** [comment]
Fixed by commit 59b346d2e50d192407f12b1402ba7548f142f758
